vault/resource_pki_secret_backend_intermediate_cert_request.go

```diff
@@ -183,6 +183,13 @@ func pkiSecretBackendIntermediateCertRequestResource() *schema.Resource {
 				ForceNew:      true,
 				ConflictsWith: []string{"managed_key_name"},
 			},
+			"add_basic_constraints": {
+				Type:        schema.TypeBool,
+				Description: "Set 'CA: true' in basic constraints to the CSR to allow signing with ADCS.",
```

Comment:
nit: may want to unpack what `ADCS`, also may want mention that is a really specific configuration, similar to what the API docs have.

Reply:
Actually, looking back at the community PR again, the two recent comments mention wanting this for Google CAS and not AD CS. Maybe we shouldn't name a specific one?

In that case, would you have a specific way you want this stated?

Reply:
> Actually, looking back at the community PR again, the two recent comments mention wanting this for Google CAS and not AD CS. Maybe we shouldn't name a specific one?

Ideally our docs would closely align with https://developer.hashicorp.com/vault/api-docs/secret/pki#add_basic_constraints. Especially the "workaround" bit.

> 
> In that case, would you have a specific way you want this stated?

https://developer.hashicorp.com/vault/api-docs/secret/pki#add_basic_constraints seems to suggest that the field provides a "workaround" for ADCS. Perhaps we need to update the API docs as well?